recipes/cmake/combined/conanfile.py

```diff
@@ -0,0 +1,197 @@
+import os
+
+from conan import ConanFile
+from conan.tools.files import get, copy, rmdir
+from conan.tools.cmake import CMake, CMakeToolchain, cmake_layout, CMakeDeps
+from conan.tools.build import cross_building, check_min_cppstd
+from conan.tools.scm import Version
+from conan.tools.microsoft import is_msvc
+from conan.errors import ConanInvalidConfiguration
+
+required_conan_version = ">=1.51.0"
+
+
+class CMakeConan(ConanFile):
+    name = "cmake"
+    package_type = "application"
+    description = "CMake, the cross-platform, open-source build system."
+    topics = ("build", "installer")
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/Kitware/CMake"
+    license = "BSD-3-Clause"
+    settings = "os", "arch", "compiler", "build_type"
+
+    options = {
+        "with_openssl": [True, False],
+        "from_sources": [True, False],
+    }
+    default_options = {
+        "with_openssl": True,
+        "from_sources": False,
+    }
+
+    def set_name(self):
+        self.name = os.environ.get('CMAKE_RECIPE_NAME', 'cmake')
+        print(self.name)
```

Comment:
```suggestion
```
Is this required?

Reply:
This is a bit out of scope for the role of ConanCenter. This is a very niche problem which the majority of users do not face so it's hard to justify the additional complexity.

CMake being such a well known tool, it's often used as an example and this is not something we'd recommend. 

> if we somehow build a broken one and store it

I would recommend improving your testing, the `--test-folder` is there to give you this flexibility, as a first step rather then having a separate name just to

> we don't have to do it again

This is was lockfiles are perfect for so I'd encourage you to look at your process :)

Reply:
I'll take `set_name` out then, no biggie. 

> I would recommend improving your testing

True, we don't _upload_ anything that's broken...except that if we say "use the latest cmake as a `tool_requires` to build cmake", then build a broken one, it's in the _local_ cache, fails tests, tell the automation to build it again, and it tries to build CMake with a version of itself that it just built that doesn't work...It's a bootstrapping problem, and you need something stable if the "build yourself with yourself" loop fails.

CMake is unusual in the sense that it's so critical to building almost everything else, and is best built with another copy of itself.

But your commentary makes me think maybe I can rethink the process...it's amazing what fresh insights come several months after the release of solution 1.x. Thanks.

(As an aside, I'm _so old_, that I remember getting GCC from a series of postings on `alt.sources.something` on Usenet, compiling it with the K&R compiler on a machine, then compiling it with itself, then compiling it with itself for a second time, checking to make sure the second and third builds produce the same result. The analogue of the K&R compiler is building the Conan CMake package with a CMake that was bootstrapped with autotools, or a carefully stored "known good" Conan CMake package.)

Reply:
❤️